### PR TITLE
Feature all delete api

### PR DESCRIPTION
**Pull request by purisaurabh**
This Feature Branch Contains :
      1) All Delete_APIs for Profile_Builder Project
      2) Test Cases for all delete apis

### CODE REVIEW

File: internal/api/handler/achievement_handler.go [hunk 1]
```diff
@@ -138,3 +134,35 @@ func ListAchievementsHandler(ctx context.Context, achSvc service.Service) func(h
...
+			}
+			middleware.ErrorResponse(w, http.StatusBadGateway, errors.ErrFailedToDelete)
+			zap.S().Error("error while deleting the achievements: ", err)
+			return
+		}
+
+		middleware.SuccessResponse(w, http.StatusOK, specs.MessageResponse{
+			Message: "Achievement deleted successfully",
+		})
+	}
```
Comment (OmkeshJosh) [inline internal/api/handler/achievement_handler.go:165]: Add log here with profile ID and achievement ID

File: internal/api/handler/achievement_handler.go [hunk 2]
```diff
@@ -138,3 +134,35 @@ func ListAchievementsHandler(ctx context.Context, achSvc service.Service) func(h
...
+			return
+		}
+		// Decode into the struct
+		req := decodeDeleteAchievementRequest(profileID, achID)
+
+		// call the service
+		err = achSvc.DeleteAchievement(ctx, req)
+		if err != nil {
+			if err == errors.ErrNoData {
+				middleware.SuccessResponse(w, http.StatusOK, specs.MessageResponse{
```
Comment (OmkeshJosh) [inline internal/api/handler/achievement_handler.go:152]: I think no need to return an error if record not found while deletion, 
just log the warning msg no record found.

@Payal13 what you think?

File: internal/pkg/specs/achievement.go [hunk 3]
```diff
@@ -74,3 +74,8 @@ func (req *UpdateAchievementRequest) Validate() error {
 
 	return nil
 }
+
+type DeleteAchievementRequest struct {
```
Comment (OmkeshJosh) [inline internal/pkg/specs/achievement.go:78]: No need to define struct for deletion oration, 

Service method signature like : 

`DeleteAchievement(ctx context.Context, profileID, achievementID int)`


same for all

Comment (purisaurabh) [reply internal/pkg/specs/achievement.go:78]: done

File: internal/api/handler/achievement_handler.go (hunk 1 above)
Comment (purisaurabh) [reply internal/api/handler/achievement_handler.go:165]: done

File: internal/api/handler/achievement_handler.go (hunk 2 above)
Comment (purisaurabh) [reply internal/api/handler/achievement_handler.go:152]: discussed with payal they told to keep as it but change the message

File: internal/pkg/constants/app.go [hunk 4]
```diff
@@ -149,3 +149,8 @@ var (
 	DefaultHealthCheck    time.Duration = 1 * 60  // 60 seconds
 	DefaultConnectTimeout time.Duration = 5       // 5 seconds
 )
+
+// Constant Message
+var (
+	NoResourceFound = "Resource with requested ID not found"
```
Comment (OmkeshJosh) [inline internal/pkg/constants/app.go]: ResourceNotFound = "Resource not found for the given request ID"

Comment (purisaurabh) [reply internal/pkg/constants/app.go]: Done

**Comment by OmkeshJosh** [comment]
LGTM

**Pull request merged**